glean-core/csharp/Glean/Dispatchers.cs

```diff
@@ -289,5 +210,135 @@ public static void AssertInTestingMode()
         {
             Debug.Assert(TestingMode);
         }
+
+        // Provides a task scheduler that ensures a maximum concurrency level while
+        // running on top of the thread pool.  Copied from example provided here:
+        // https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.taskscheduler?view=netcore-3.1
+        private class LimitedConcurrencyLevelTaskScheduler : TaskScheduler
```

Comment:
Would you kindly move this to a separate file?  It's also a bit unclear if we're allowed to use this and what's the license for this.

Reply:
Sure, that makes it a little more open within the project, but I can put it under the Dispatchers namespace to help with that.  [The original code sample](https://docs.microsoft.com/en-us/samples/dotnet/samples/parallel-programming-extensions-extras-cs/) uses the MIT license, which I think is compatible for us to use?

Reply:
> This will change once we'll add the metrics ping scheduler. That's why we need to get this right. I'm fine with not keeping this PR waiting, we should file a bug and deal with this specific aspect later.

[Bug filed](https://bugzilla.mozilla.org/show_bug.cgi?id=1646750)